okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -186,6 +189,9 @@ private boolean resolveRedirects(ProxyContext pc,
   private List<ModuleInstance> getModulesForRequest(ProxyContext pc,
                                                     List<ModuleDescriptor> enabledModules) {
```

Comment:
This is the slowest method reported in OKAPI-868.
Can you put in more granular metrics? For example
* `ProxyService.getModulesForRequest for enabledModules`
* `ProxyService.getModulesForRequest for rr 1`
* `ProxyService.getModulesForRequest for rr 2`
* `ProxyService.getModulesForRequest sort`
* `ProxyService.getModulesForRequest for mods`

Reply:
Thanks @julianladisch. Looks like all slowness of `getModulesForRequest` are from `moduleManager.getEnabledModules`. I updated the [OKAPI-868](https://issues.folio.org/browse/OKAPI-868) with perf report [link](http://carrier-io.int.folio.ebsco.com/grafana/d/b3iP-8hiz/java-profiling-folio-new?orgId=1&from=1601352757170&to=1601354777273) and relevant stacktraces screenshots. Please take a look the report and let me know if you feel strongly we should add more granular metrics for `getModulesForRequest` calls.

Reply:
Never mind. I see what you mean. It is not about callers, but parts within `getModulesForRequest` itself.

Reply:
@julianladisch, I added granular metrics as suggested.